tests/test_ignore_patterns.py

```
@@ -0,0 +1,327 @@
+
+    def test_respects_gitignore_when_no_tldrignore(self, tmp_path: Path):
+        """Should respect .gitignore patterns when .tldrignore is missing."""
+        import subprocess
+        from tldr.api import get_code_structure
+
+        # Setup: Create project with only .gitignore (no .tldrignore)
+        project = tmp_path / "project"
+        project.mkdir()
+
+        # Initialize as git repo (required for gitignore checking)
+        subprocess.run(["git", "init"], cwd=project, capture_output=True, check=True)
```

Ruff (0.14.14)

[error] Starting a process with a partial executable path

(S607)